pkg/sink/sink.go

```diff
@@ -91,17 +95,40 @@ type Response struct {
 	ErrorMessage string `json:"errorMessage,omitempty"`
 }
 
+func (r Sink) emitEvents(recorder record.EventRecorder, el *triggersv1.EventListener, err error) {
+	afterCondition := el.Status.GetCondition(apis.ConditionReady)
+	if el.Annotations[events.EnableEventListenerEvents] == "true" {
+		events.Emit(recorder, afterCondition, el, err)
+	}
+}
+
 // HandleEvent processes an incoming HTTP event for the event listener.
 func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 	log := r.Logger.With(
 		zap.String("eventlistener", r.EventListenerName),
 		zap.String("namespace", r.EventListenerNamespace),
 	)
+	elTemp := triggersv1.EventListener{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "EventListener",
+			APIVersion: "v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.EventListenerName,
+			Namespace: r.EventListenerNamespace,
+			Annotations: map[string]string{
+				// enabled by default for temporary EL
+				events.EnableEventListenerEvents: "true",
+			}}}
+
+	r.emitEvents(r.EventRecorder, &elTemp, nil)
```

Comment:
Doesn't this mean that we will always emit an event due to `elTemp` always setting the annotation to true? i.e. the actual EL could have it set of false but due to `elTemp` defaulting to true, we are still emitting one event?

Reply:
Yes because from this operation
`el, err := r.EventListenerLister.EventListeners(r.EventListenerNamespace).Get(r.EventListenerName)` there could be chances of error while doing Get 
So added elTemp so that get operation failure captured